### PR TITLE
Implemented api call throttling

### PR DESCRIPTION
**Pull request by Ilithor**

### CODE REVIEW

File: src/components/like/LikeButton.js [hunk 1]
```diff
@@ -15,19 +15,32 @@ import { getLikePost } from '../../redux/actions/dataActions';
...
  */
 export const LikeButton = ({ postId, getLikePost, UI = {} }) => {
+  const makeCall = useRef(false);
   const likePost = () => {
+    if (!!makeCall.current) return;
+    makeCall.current = true;
     getLikePost(postId);
+    setTimeout(() => {
+      makeCall.current = false;
+    }, 3000);
```
Comment (jeffersoneagley) [inline src/components/like/LikeButton.js]: this is a clever solution. On thinking about it though, shouldn't this logic be something more like:
```
if(!inProgress){
  inProgress:true
  axios.put('post/',{whatever data blabla})
  .then((data)=>{
    inProgress:false
    whatever logic with that data
})
}
```

Comment (jeffersoneagley) [inline src/components/like/LikeButton.js]: this is a clever solution. On thinking about it though, shouldn't this logic be something more like:
```
if(!inProgress){
  inProgress:true
  axios.put('post/',{whatever data blabla})
  .then((data)=>{
    whatever logic with that data
  })
  .catch(ex=>{
    whatever logic with that error
  })
  .finally(()=>{
    inProgress:false
  })
}
```

Comment (Ilithor) [reply src/components/like/LikeButton.js]: As the point of the function is calling an action function to alter data in the store, there isn't any return data. I'm also trying to keep axios calls off of my components as much as possible, and keeping them to the fetch file, as I find axios calls to look a little messy.

Comment (jeffersoneagley) [reply src/components/like/LikeButton.js]: I usually put axios/odata/fetch/whatever calls in their own file, where each getWhateverData returns a Promise<WhateverDataType>, then in the action creator
```
if(!inProgress){
  inProgress:true
  getWhateverData()
  .then((data)=>{
    whatever logic with that data
  })
  .catch(ex=>{
    whatever logic with that error
  })
  .finally(()=>{
    inProgress:false
  })
}
```

**Pull request merged**